app/controllers/application_controller.rb

```diff
@@ -111,6 +111,9 @@ def store_interaction
       # as of Rack 2.0.8, the session_id is wrapped in a class of its own
       # it is not a string anymore
       # see https://github.com/rack/rack/issues/1433
+
+      return if request.session_options[:id].nil?
```

Comment:
It turned out that in the newly added controller tests for the vouchers controller, `request.session_options[:id ]` was nil, so an exception was raised. I introduced this line to get rid of the exception.

Reply:
Ok, I can reproduce that the tests are failing if we remove the line. However, I'm not quite sure I understand how this `store_interaction` mechanism works (the method where you added the line). Do we log every single interaction (i.e. request that the browser of the user makes) via a Sidekiq worker in the interactions database? If so, why was this introduced (to prevent abuse of MaMpf?) and what's its relationship to GDPR?

Reply:
discussed off-GitHub, it is compliant